### ISSUE TITLE
Extract functions to separate CommonJS modules

### ISSUE THREAD

**Issue by esamattis** [esamattis/underscore.string#355]
Creating new issue from #354.

Each function will be split into a separate CommonJS module like in [101](https://github.com/tjmehta/101/)

This will mean few things
- Browserify users can use only parts of this library by calling `var slugify = require("underscore.string/slugify")`
- No changes for users who want to the full library like before
- Contributing will become easier since less merge conflicts will occur between pull requests touching different functions

The work has already begun in the commonjs branch

**Comment by epeli** [comment]
http://blog.andyet.com/2015/01/07/modularizing-underscorejs and https://github.com/ampersandjs/amp

**Comment by stoeffel** [comment]
>  modularaize underscorejs

This sounds great. Maybe we should consider this for u.string too, since we just separated the modules.
The build task could create separate modules and update their version number if they have changed. So we could publish u.string as a complete lib and as separated namespaced modules.

**Comment by epeli** [comment]
I think creating folder for each is kinda overkill but it's interesting to see how it pans out for Amp

**Comment by stoeffel** [comment]
Yes, you are probably right. I think we made a create improvement with splitting into modules. Maybe we could come back to this idea in the future, but for now it is not necessary.

**Comment by jdalton** [comment]
Just a heads up Lo-Dash is moving into string methods [a bit](https://github.com/lodash/lodash/tree/npm/string)-[too](https://github.com/lodash/lodash/tree/npm-packages) for its v3.0. Accessible via `require('lodash/string/trunc')` or `require('lodash.trunc')` or as [AMD](https://github.com/lodash/lodash/tree/amd/string) or as ES6 modules.

**Comment by epeli** [comment]
Interesting discussions here https://news.ycombinator.com/item?id=8853356

@jdalton that's cool. I'm glad to see method specific requires in lodash too. Didn't notice it earlier.

**Comment by jdalton** [comment]
Before v3.0 they were in `lodash-node` but paths were longer to type as it contained both `compat` and `modern` builds. With v3.0 we're going the `lodash` is modern with a `lodash-compat` package for older browsers. This also aligns with jQuery v3.0 which will have `jquery` and `jquery-compat`.

As for modules. We've had individual npm packages, like `lodash.uniq`, and module bundles since v2.0. We got push back though for being too modular (some devs didn't like the dep graph). So with v3.0 we've improved our ability to inline modules so the v3.0 individual npm packages have much smaller dep graphs at the cost of some code dup. The module bundled in the primary package have no inlining as it's not an issue there.
